### PR TITLE
much sought-after marathon docs

### PR DESCRIPTION
**Pull request by sascala**
## What type of changes does your doc introduce?
- [ ] Bug fix
- [x] New feature 
## What is the urgency level of this change?
- [ ] Blocker
- [x] High
- [ ] Medium
## I have completed these items:
- [x] I have tested all commands and code snippets.
- [x] I have built locally and tested for broken links and formatting.
- [x] I have made this change for all affected versions (e.g. 1.7, 1.8, and 1.9).
- [x] My doc follows the style guide: https://github.com/dcos/dcos-docs#contributing.

Ping @emanic @sascala @joel-hamill for reviewing pull request changes.

### CODE REVIEW

File: 1.9/usage/marathon/application-basics.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+In this example, we are going to use the Marathon REST API to deploy the app `basic-3` from the DC/OS CLI:
+
+```sh
+ curl -H "Authorization: token=$(dcos config show core.dcos_acs_token) -X POST <dcos-url>/service/marathon/v2/apps -d @basic-3.json -H "Content-type: application/json"
+```
+
+When you submit the above application definition to DC/OS, you should see your new service running from the **Services** tab of the DC/OS GUI.
+
```
Comment (sascala) [inline 1.9/usage/marathon/application-basics.md]: @pyronicide I've ported this document from the Marathon site and modified it for DC/OS. The last piece of the puzzle is this final instruction -- do you happen to know where exactly a user should go to see the contents of the container's root directory?

File: 1.7/usage/marathon/application-basics.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+```bash
+dcos marathon app add <your-service-name>.json
+```
+
+When you define and launch a service, Marathon hands over execution to Mesos. Mesos creates a sandbox directory for each task. The sandbox directory is a directory on each agent node that acts as an execution environment and contains relevant log files. The `stderr` and `stdout` streams are also written to the sandbox directory.
+
+## Using Resources in Applications
+
+To run any non-trivial application, you typically depend on a collection of resources: files and/or archives of files. To manage resource allocation, Marathon has the concept of URIs. URIs use the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
```
Comment (joel-hamill) [inline 1.7/usage/marathon/application-basics.md]: Avoid the `and/or` construction. (https://americaneditor.wordpress.com/2013/03/11/the-drama-of-and-and-or/)

File: 1.8/usage/marathon/application-basics.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+```bash
+dcos marathon app add <your-service-name>.json
+```
+
+When you define and launch a service, Marathon hands over execution to Mesos. Mesos creates a sandbox directory for each task. The sandbox directory is a directory on each agent node that acts as an execution environment and contains relevant log files. The `stderr` and `stdout` streams are also written to the sandbox directory.
+
+## Using Resources in Applications
+
+To run any non-trivial application, you typically depend on a collection of resources: files and/or archives of files. To manage resource allocation, Marathon has the concept of URIs. URIs use the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
```
Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: - `URI (uniform resource identifier)`
- Whats the Mesos fetcher?

File: 1.8/usage/marathon/application-basics.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    "cpus": 0.1,
+    "mem": 10.0,
+    "instances": 1,
+    "uris": [
+        "https://example.com/app/cool-script.sh"
+    ]
+}
+```
+
+What the example above does: before executing the `cmd`, download the resource `https://example.com/app/cool-script.sh` (via Mesos) and make it available in the service instance's sandbox. Verify that it had been downloaded by visiting the DC/OS web interface and clicking on an instance of `basic-1`, then on the **Files** tab. You should find `cool-script.sh` there.
```
Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: `service instance's Mesos sandbox`?

Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: This sentence should use active voice (e.g. `The example above executes the `cmd`...`

File: 1.8/usage/marathon/application-basics.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    "instances": 1,
+    "uris": [
+        "https://example.com/app.zip"
+    ]
+}
+```
+
+In contrast to the example `basic-1` we now have a `cmd` with the value `app/cool-script.sh`. When the zip file gets downloaded and extracted, a directory `app` according to the file name `app.zip` is created and the content of the zip file is extracted into it.
+
+You can specify more than one resource. For example, you could provide a git repository and some resources from a CDN as follows:
```
Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: `Git`

File: 1.8/usage/marathon/application-basics.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+* `hdfs:`
+* `s3:`
+* `s3a:`
+* `s3n:`
+
+## A Simple Docker-based Application
+
+With Marathon it is straightforward to run applications that use Docker images.
+
+In the following example application definition, we will focus on a simple Docker app: a Python-based web server using the image [python:3](https://registry.hub.docker.com/_/python/). Inside the container, the web server runs on port `8080` (the value of `containerPort`). `hostPort` is set to `0` so that Marathon assigns a random port on the Mesos agent, which is mapped to port 8080 inside the container.
```
Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: Rewrite to explain that users are supposed to create an app named `basic-3.json`, and then deploy to DCOS using an API command. This is pretty hard to follow as-is.

File: 1.8/usage/marathon/application-basics.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+      ]
+    }
+  }
+}
+```
+
+In this example, we are going to use the Marathon REST API to deploy the app `basic-3` from the DC/OS CLI:
+
+```sh
+ curl -H "Authorization: token=$(dcos config show core.dcos_acs_token) -X POST <dcos-url>/service/marathon/v2/apps -d @basic-3.json -H "Content-type: application/json"
```
Comment (joel-hamill) [inline 1.8/usage/marathon/application-basics.md]: `curl -H "Authorization: token=$(dcos config show core.dcos_acs_token)"` - missing closing quotes

File: 1.7/usage/marathon/application-basics.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
+---
+post_title: Application Definitions
+nav_title: Application Definitions
+menu_order: 0
+---
+
+Applications are an integral concept in Marathon. Each application typically represents a long-running service, of which there would be many instances running on multiple hosts. An application instance is called a *task*. The *application definition* describes everything needed to start and maintain the tasks. A Marathon application definition creates a DC/OS _service_.
```
Comment (joel-hamill) [inline 1.7/usage/marathon/application-basics.md]: `Applications are an integral concept in DC/OS`

File: 1.7/usage/marathon/application-basics.md (hunk 2 above)
Comment (sascala) [inline 1.7/usage/marathon/application-basics.md]: 👍

File: 1.8/usage/marathon/application-basics.md (hunk 3 above)
Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: 👍

File: 1.8/usage/marathon/application-basics.md (hunk 4 above)
Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: :+1:

Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: yes. I -- ahem -- didn't write the original of this.

File: 1.8/usage/marathon/application-basics.md (hunk 5 above)
Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: 👍

File: 1.8/usage/marathon/application-basics.md (hunk 6 above)
Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: done

File: 1.8/usage/marathon/application-basics.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+When you submit the above application definition to DC/OS, you should see your new service running from the **Services** tab of the DC/OS GUI.
+
+Go to `<dcos-url>:3100` to see the contents of the Docker container's root directory.
```
Comment (sascala) [inline 1.8/usage/marathon/application-basics.md]: OK. I'll make a Jira
